src/npc.cpp

```diff
@@ -327,8 +348,10 @@ void Npc::onThink(uint32_t interval)
 		m_npcEventHandler->onThink();
 	}
 
-	if (getTimeSinceLastMove() >= walkTicks) {
-		addEventWalk();
+	if (!isIdle) {
+		if (getTimeSinceLastMove() >= walkTicks) {
```

Comment:
is there a reason here for using nested ifs? If not, then move the second conditions into the first if statement.